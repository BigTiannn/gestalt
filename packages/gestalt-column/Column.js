/* @flow */
import React, { PropTypes } from 'react';

type Columns = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;

export default function Column({ children, span = 12 }: { children?: any, span: Columns }) {
  return <div className={`col-${span}`}>{children}</div>;
}

Column.propTypes = {
  children: PropTypes.node,
  span: PropTypes.number.isRequired,
};
