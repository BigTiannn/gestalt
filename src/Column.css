:root {
  --ncols: 12;
}

:global .col {
  display: inline-block;
  vertical-align: top;
}

:global .col-1 {
  width: calc(1 / var(--ncols) * 100%);
}

:global .col-2 {
  width: calc(2 / var(--ncols) * 100%);
}

:global .col-3 {
  width: calc(3 / var(--ncols) * 100%);
}

:global .col-4 {
  width: calc(4 / var(--ncols) * 100%);
}

:global .col-5 {
  width: calc(5 / var(--ncols) * 100%);
}

:global .col-6 {
  width: calc(6 / var(--ncols) * 100%);
}

:global .col-7 {
  width: calc(7 / var(--ncols) * 100%);
}

:global .col-8 {
  width: calc(8 / var(--ncols) * 100%);
}

:global .col-9 {
  width: calc(9 / var(--ncols) * 100%);
}

:global .col-10 {
  width: calc(10 / var(--ncols) * 100%);
}

:global .col-11 {
  width: calc(11 / var(--ncols) * 100%);
}

:global .col-12 {
  width: calc(12 / var(--ncols) * 100%);
}

@media (--sm) {
  :global .sm-col-1 {
    width: calc(1 / var(--ncols) * 100%);
  }

  :global .sm-col-2 {
    width: calc(2 / var(--ncols) * 100%);
  }

  :global .sm-col-3 {
    width: calc(3 / var(--ncols) * 100%);
  }

  :global .sm-col-4 {
    width: calc(4 / var(--ncols) * 100%);
  }

  :global .sm-col-5 {
    width: calc(5 / var(--ncols) * 100%);
  }

  :global .sm-col-6 {
    width: calc(6 / var(--ncols) * 100%);
  }

  :global .sm-col-7 {
    width: calc(7 / var(--ncols) * 100%);
  }

  :global .sm-col-8 {
    width: calc(8 / var(--ncols) * 100%);
  }

  :global .sm-col-9 {
    width: calc(9 / var(--ncols) * 100%);
  }

  :global .sm-col-10 {
    width: calc(10 / var(--ncols) * 100%);
  }

  :global .sm-col-11 {
    width: calc(11 / var(--ncols) * 100%);
  }

  :global .sm-col-12 {
    width: calc(12 / var(--ncols) * 100%);
  }
}

@media (--md) {
  :global .md-col-1 {
    width: calc(1 / var(--ncols) * 100%);
  }

  :global .md-col-2 {
    width: calc(2 / var(--ncols) * 100%);
  }

  :global .md-col-3 {
    width: calc(3 / var(--ncols) * 100%);
  }

  :global .md-col-4 {
    width: calc(4 / var(--ncols) * 100%);
  }

  :global .md-col-5 {
    width: calc(5 / var(--ncols) * 100%);
  }

  :global .md-col-6 {
    width: calc(6 / var(--ncols) * 100%);
  }

  :global .md-col-7 {
    width: calc(7 / var(--ncols) * 100%);
  }

  :global .md-col-8 {
    width: calc(8 / var(--ncols) * 100%);
  }

  :global .md-col-9 {
    width: calc(9 / var(--ncols) * 100%);
  }

  :global .md-col-10 {
    width: calc(10 / var(--ncols) * 100%);
  }

  :global .md-col-11 {
    width: calc(11 / var(--ncols) * 100%);
  }

  :global .md-col-12 {
    width: calc(12 / var(--ncols) * 100%);
  }
}

@media (--lg) {
  :global .lg-col-1 {
    width: calc(1 / var(--ncols) * 100%);
  }

  :global .lg-col-2 {
    width: calc(2 / var(--ncols) * 100%);
  }

  :global .lg-col-3 {
    width: calc(3 / var(--ncols) * 100%);
  }

  :global .lg-col-4 {
    width: calc(4 / var(--ncols) * 100%);
  }

  :global .lg-col-5 {
    width: calc(5 / var(--ncols) * 100%);
  }

  :global .lg-col-6 {
    width: calc(6 / var(--ncols) * 100%);
  }

  :global .lg-col-7 {
    width: calc(7 / var(--ncols) * 100%);
  }

  :global .lg-col-8 {
    width: calc(8 / var(--ncols) * 100%);
  }

  :global .lg-col-9 {
    width: calc(9 / var(--ncols) * 100%);
  }

  :global .lg-col-10 {
    width: calc(10 / var(--ncols) * 100%);
  }

  :global .lg-col-11 {
    width: calc(11 / var(--ncols) * 100%);
  }

  :global .lg-col-12 {
    width: calc(12 / var(--ncols) * 100%);
  }
}
