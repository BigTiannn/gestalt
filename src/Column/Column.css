:root {
  --ncols: 12;
}

.col {
  display: inline-block;
  vertical-align: top;
}

.col-1 {
  width: calc(1 / var(--ncols) * 100%);
}

.col-2 {
  width: calc(2 / var(--ncols) * 100%);
}

.col-3 {
  width: calc(3 / var(--ncols) * 100%);
}

.col-4 {
  width: calc(4 / var(--ncols) * 100%);
}

.col-5 {
  width: calc(5 / var(--ncols) * 100%);
}

.col-6 {
  width: calc(6 / var(--ncols) * 100%);
}

.col-7 {
  width: calc(7 / var(--ncols) * 100%);
}

.col-8 {
  width: calc(8 / var(--ncols) * 100%);
}

.col-9 {
  width: calc(9 / var(--ncols) * 100%);
}

.col-10 {
  width: calc(10 / var(--ncols) * 100%);
}

.col-11 {
  width: calc(11 / var(--ncols) * 100%);
}

.col-12 {
  width: calc(12 / var(--ncols) * 100%);
}

@media (--sm) {
  .sm-col-1 {
    width: calc(1 / var(--ncols) * 100%);
  }

  .sm-col-2 {
    width: calc(2 / var(--ncols) * 100%);
  }

  .sm-col-3 {
    width: calc(3 / var(--ncols) * 100%);
  }

  .sm-col-4 {
    width: calc(4 / var(--ncols) * 100%);
  }

  .sm-col-5 {
    width: calc(5 / var(--ncols) * 100%);
  }

  .sm-col-6 {
    width: calc(6 / var(--ncols) * 100%);
  }

  .sm-col-7 {
    width: calc(7 / var(--ncols) * 100%);
  }

  .sm-col-8 {
    width: calc(8 / var(--ncols) * 100%);
  }

  .sm-col-9 {
    width: calc(9 / var(--ncols) * 100%);
  }

  .sm-col-10 {
    width: calc(10 / var(--ncols) * 100%);
  }

  .sm-col-11 {
    width: calc(11 / var(--ncols) * 100%);
  }

  .sm-col-12 {
    width: calc(12 / var(--ncols) * 100%);
  }
}

@media (--md) {
  .md-col-1 {
    width: calc(1 / var(--ncols) * 100%);
  }

  .md-col-2 {
    width: calc(2 / var(--ncols) * 100%);
  }

  .md-col-3 {
    width: calc(3 / var(--ncols) * 100%);
  }

  .md-col-4 {
    width: calc(4 / var(--ncols) * 100%);
  }

  .md-col-5 {
    width: calc(5 / var(--ncols) * 100%);
  }

  .md-col-6 {
    width: calc(6 / var(--ncols) * 100%);
  }

  .md-col-7 {
    width: calc(7 / var(--ncols) * 100%);
  }

  .md-col-8 {
    width: calc(8 / var(--ncols) * 100%);
  }

  .md-col-9 {
    width: calc(9 / var(--ncols) * 100%);
  }

  .md-col-10 {
    width: calc(10 / var(--ncols) * 100%);
  }

  .md-col-11 {
    width: calc(11 / var(--ncols) * 100%);
  }

  .md-col-12 {
    width: calc(12 / var(--ncols) * 100%);
  }
}

@media (--lg) {
  .lg-col-1 {
    width: calc(1 / var(--ncols) * 100%);
  }

  .lg-col-2 {
    width: calc(2 / var(--ncols) * 100%);
  }

  .lg-col-3 {
    width: calc(3 / var(--ncols) * 100%);
  }

  .lg-col-4 {
    width: calc(4 / var(--ncols) * 100%);
  }

  .lg-col-5 {
    width: calc(5 / var(--ncols) * 100%);
  }

  .lg-col-6 {
    width: calc(6 / var(--ncols) * 100%);
  }

  .lg-col-7 {
    width: calc(7 / var(--ncols) * 100%);
  }

  .lg-col-8 {
    width: calc(8 / var(--ncols) * 100%);
  }

  .lg-col-9 {
    width: calc(9 / var(--ncols) * 100%);
  }

  .lg-col-10 {
    width: calc(10 / var(--ncols) * 100%);
  }

  .lg-col-11 {
    width: calc(11 / var(--ncols) * 100%);
  }

  .lg-col-12 {
    width: calc(12 / var(--ncols) * 100%);
  }
}
