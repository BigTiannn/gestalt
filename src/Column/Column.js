/* @flow */
import React, { PropTypes } from 'react';
import Box from '../Box/Box';

type Columns = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;

type Props = {
  children?: any,
  xs: Columns,
  sm?: Columns,
  md?: Columns,
  lg?: Columns,
};

export default function Column(props: Props) {
  const { children, xs, sm, md, lg } = props;
  return (
    <Box
      xs={{ display: 'inlineBlock', column: xs }}
      sm={{ column: sm }}
      md={{ column: md }}
      lg={{ column: lg }}
      dangerouslySetInlineStyle={{ __style: { verticalAlign: 'top' } }}
    >
      {children}
    </Box>
  );
}

Column.propTypes = {
  children: PropTypes.node,
  xs: PropTypes.number.isRequired,
  sm: PropTypes.number,
  md: PropTypes.number,
  lg: PropTypes.number,
};
