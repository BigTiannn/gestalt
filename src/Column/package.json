{
  "name": "gestalt-column",
  "version": "0.5.0",
  "main": "./Column.js",
  "dependencies": {
    "react": "^15.3.1",
    "classnames": "^2.2.5"
  }
}
