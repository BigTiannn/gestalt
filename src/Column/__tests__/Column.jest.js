/* eslint-env jest */
/* eslint import/imports-first: 0 */
jest.unmock('../Column');

import React from 'react';
import renderer from 'react-test-renderer';
import Column from '../Column';

test('Column matches snapshot', () => {
  const component = renderer.create(
    <Column xs={1}>Hello world</Column>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
