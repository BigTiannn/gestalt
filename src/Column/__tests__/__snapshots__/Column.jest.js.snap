exports[`test Column matches snapshot 1`] = `
<div
  className="col col-1">
  Hello world
</div>
`;
